File: Arrays & Hashing/Java/contains_duplicates.java
import java.util.HashSet; 
import java.util.Set;

class Solution {
    public boolean hasDuplicate(int[] nums) {
        Set<Integer> tracker = new HashSet<>(); 
        for (int num : nums) {
            if (tracker.contains(num))
                return true;
            tracker.add(num);
        }
        return false;
    }
}

File: Arrays & Hashing/cpp/contains_duplicates.cpp
#include <vector>
#include <unordered_set>

class Solution {
public:
    bool hasDuplicate(std::vector<int>& nums) {
        std::unordered_set<int> tracker; 
        for (int num : nums) {
            if (tracker.count(num)) {
                return true; 
            }
            tracker.insert(num); 
        }
        return false; 
    }
};
